app/src/main/java/org/mozilla/fenix/components/metrics/ActivationPing.kt

```diff
@@ -0,0 +1,161 @@
+    internal fun markAsTriggered() {
+        prefs.edit().putBoolean("ping_sent", true).apply()
+    }
+
+    /**
+     * Query the Google Advertising API to get the Google Advertising ID.
+     *
+     * This is meant to be used off the main thread. The API will throw an
+     * exception and we will print a log message otherwise.
+     *
+     * @return a String containing the Google Advertising ID or null.
+     */
+    @Suppress("TooGenericExceptionCaught")
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun getAdvertisingID(): String? {
+        return try {
+            AdvertisingIdClient.getAdvertisingIdInfo(context).id
+        } catch (e: GooglePlayServicesNotAvailableException) {
+            Logger.debug("ActivationPing - Google Play not installed on the device")
+            null
+        } catch (e: GooglePlayServicesRepairableException) {
+            Logger.debug("ActivationPing - recoverable error connecting to Google Play Services")
+            null
+        } catch (e: IllegalStateException) {
+            // This is unlikely to happen, as this should be running off the main thread.
+            Logger.debug("ActivationPing - AdvertisingIdClient must be called off the main thread")
+            null
+        } catch (e: IOException) {
+            Logger.debug("ActivationPing - unable to connect to Google Play Services")
+            null
+        } catch (e: NullPointerException) {
+            Logger.debug("ActivationPing - no Google Advertising ID available")
+            null
+        }
+    }
+
+    /**
+     * Get the salt to use for hashing. This is a convenience
+     * function to help with unit tests.
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun getHashingSalt(): String = context.applicationInfo.packageName
+
+    /**
+     * Produces an hashed version of the Google Advertising ID.
+     * We want users using more than one of our products to report a different
+     * ID in each of them. This function runs off the main thread and is CPU-bound.
+     *
+     * @return an hashed and salted Google Advertising ID or null if it was not possible
+     *         to get the Google Advertising ID.
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal suspend fun getHashedIdentifier(): String? = withContext(Dispatchers.Default) {
+        getAdvertisingID()?.let { unhashedID ->
+            // Add some salt to the ID, before hashing. For this specific use-case, it's ok
+            // to use the same salt value for all the hashes. We want hashes to be stable
+            // within a single product, but we don't want hashes to be the same across different
+            // products (e.g. Fennec vs Fenix).
+            val salt = getHashingSalt()
+
+            // Apply hashing.
+            try {
+                val saltedID = unhashedID + salt
+                val digest = MessageDigest.getInstance("SHA-256")
```

Comment:
Is there any chance we could use something else other than this? Is there anything else available for us to use (we really want `bcrypt` here) in the Fenix repo or Android API?